### PR TITLE
Standardized Date and Time

### LINKED ISSUE #5490: Date and Time: Standardize it across system
The date and time are using different formats in the system. Please identify different places for the date and time usage and standardize the date and time in all ways.

### PR DESCRIPTION
**Pull request by Saumyachitransh01**
Fixes #5490 

#### Short description of what this resolves:

All the occurrence of dates and time has been converted to only 5 formats 

#### Changes proposed in this pull request:

Since there were more than 20 formats used all over the platform, I've minimized it to use only 5 formats.
Five formats that are used everywhere including all .js files and .hbs files are:-
1. DD-MM-YYYY
2. ddd, DD MMMM, YYYY
3. hh:mm A
4. ddd, DD MMMM, YYYY hh:mm A
5. DD MMMM, YYYY hh:mm A

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/components/forms/wizard/basic-details-step.js [hunk 1]
```diff
@@ -116,10 +116,10 @@ export default Component.extend(FormMixin, EventWizardMixin, {
     this.oldTimezone = this.data.event.timezone;
     if (!oldTimezone || !this.oldTimezone || oldTimezone === this.oldTimezone) {return}
     if (event.startsAt) {
-      event.startsAt = moment.tz(event.startsAt.clone().tz(oldTimezone).format('YYYY-MM-DDTHH:mm:ss.SSS'), moment.ISO_8601, this.data.event.timezone);
+      event.startsAt = moment.tz(event.startsAt.clone().tz(oldTimezone).format('DD MMMM, YYYY hh:mm A'), moment.ISO_8601, this.data.event.timezone);
```
Comment (iamareebjamal) [inline app/components/forms/wizard/basic-details-step.js:119]: This is not user facing format. This is ISO8601 format and it needs to remain same

File: app/components/schedule.hbs [hunk 2]
```diff
@@ -15,10 +15,10 @@
     @viewName="agendaDay"
     @defaultView="timelineDay"
     @validRange={{this.validRange}}
-    @now={{moment-format (now) "YYYY-MM-DD"}}
+    @now={{moment-format (now) "DD-MM-YYYY"}}
```
Comment (iamareebjamal) [inline app/components/schedule.hbs:18]: This is not user facing format. This needs to be passed in the exact same way to the component

File: app/components/schedule.hbs [hunk 3]
```diff
@@ -15,10 +15,10 @@
...
     @scrollTime="0:00:00"
     @slotDuration="00:05:00"
-    @slotLabelFormat="HH:mm"
+    @slotLabelFormat="hh:mm A"
```
Comment (iamareebjamal) [inline app/components/schedule.hbs:21]: Same. This should not be changed

**Comment by Saumyachitransh01** [comment]
Okay, and apart from these three are there any other corrections needed?

**Comment by iamareebjamal** [comment]
@mariobehling Please check

**Review by mariobehling — CHANGES_REQUESTED** [review]
Thank you very much! This is a huge improvement and seeing your changes I believe it would be even better, if we could further limit this to only two formats. Here is the list below. 

Please do not add a "0" in front of single digit date or time formats.

1. This is good for tables:
Make event format as in this example: 4 Dec, 2020 5:00 PM (IST)
* [ ] table https://eventyay.com/events
* [ ] table https://eventyay.com/admin/events
* [ ] table https://eventyay.com/admin/sales
* [ ] table https://eventyay.com//admin/users/
* [ ] table event orders table /events/[event ID]/tickets/orders
* [ ] table event attebdees /events/[event ID]/tickets/attebdees
* [ ] table session overview page events/[event ID]/sessions/all

2. This is good for most public pages
Make event format as in this example: Friday, 4 December, 2020 5:00 PM (IST)
* [ ] tickets overview https://eventyay.com/my-tickets/past
* [ ] ticket order https://eventyay.com/orders/[order number here]
* [ ] events dashboard /events/[events ID]
* [ ] public events page header e/[eventpage], e.g. https://eventyay.com/e/a315169f
* [ ] public events page ticket section, e.g. https://eventyay.com/e/a315169f#tickets
* [ ] public events page calendar section
* [ ] call for speakers, e.g. https://eventyay.com/e/a315169f/cfs

3. Am I missing something important? Please share if any.

**Comment by Saumyachitransh01** [comment]
Screenshot from 2020-11-10 05-52-07

And places like these? Having the format "MM/DD/YYYY". Which one of those two formats should I use?

**Comment by iamareebjamal** [comment]
4 Dec, 2020

File: app/components/schedule.hbs [hunk 4]
```diff
@@ -18,7 +18,7 @@
     @now={{moment-format (now) "YYYY-MM-DD"}}
     @scrollTime="0:00:00"
     @slotDuration="00:05:00"
-    @slotLabelFormat="HH:mm"
+    @slotLabelFormat="HH:MM"
```
Comment (iamareebjamal) [inline app/components/schedule.hbs:21]: Again, this is not for public display.

File: app/components/forms/wizard/basic-details-step.js [hunk 5]
```diff
@@ -465,7 +465,7 @@ export default Component.extend(FormMixin, EventWizardMixin, {
       eventStartDate = moment(new Date(eventStartDate));
       this.data.event.tickets.forEach(ticket => {
         if (moment(eventStartDate).isBefore(ticket.get('salesEndsAt'))) {
-          ticket.set('salesEndsAt', moment(eventStartDate, 'MM/DD/YYYY').toDate());
+          ticket.set('salesEndsAt', moment(eventStartDate, 'DD-MMMM-YYYY').toDate());
```
Comment (iamareebjamal) [inline app/components/forms/wizard/basic-details-step.js:468]: Hmm, where did this format come from? And this is not for public display

**Comment by iamareebjamal** [comment]
Please don't change formats which are used for parsing and not for display

**Comment by Saumyachitransh01** [comment]
> Please don't change formats which are used for parsing and not for display

So should I only change hbs and not js?

**Comment by iamareebjamal** [comment]
JS contains display formats as well. And the build is failing. The initial work was much more stable. I don't know what triggered the need for editing each file one at a time. You already had another PR ready. You could just force push the changes on this branch

**Comment by iamareebjamal** [comment]
I have forced pushed the changes from other PR to this. Please delete the local branch and fetch again to continue the work.

```sh
git checkout development
git branch -D branchNew
git fetch origin branchNew
git checkout branchNew
```

**Comment by Saumyachitransh01** [comment]
> Please don't change formats which are used for parsing and not for display

It's a bit confusing, how to determine which one is for parsing and which is for display
Screenshot from 2020-11-11 17-19-58
Please confirm if this is for display or not?

**Comment by iamareebjamal** [comment]
It is a display format

**Comment by mariobehling** [comment]
Hi, would be great if this PR gets finalized. As this PR is blocking others from resolving the issue, we would close it in the upcoming days if there is no activity. You already contributed a good amount of time to this PR, so I think with some more effort you could finalize it. Keep it up!

File: app/helpers/header-date.js [hunk 6]
```diff
@@ -3,7 +3,7 @@ import moment from 'moment';
 
 export function headerDate(params) {
   const timezone = params[1] ? params[1] : moment.tz.guess();
-  return `${moment(params[0]).tz(timezone).format('dddd, D MMMM, YYYY h:mm A')} (${moment.tz(params[0], timezone).zoneAbbr()})`;
+  return `${moment(params[0]).tz(timezone).format('dddd, MMMM Do YYYY, h:mm A')} (${moment.tz(params[0], timezone).zoneAbbr()})`;
```
Comment (iamareebjamal) [inline app/helpers/header-date.js]: Why reverted the format here? We need same format everywhere

Comment (Saumyachitransh01) [reply app/helpers/header-date.js]: This was the only way for the build to be successful.
Changing it was failing the build every time.

Comment (iamareebjamal) [reply app/helpers/header-date.js]: No, that's not the only way to fix the build

**Comment by mariobehling** [comment]
This has been resolved already elsewhere now. Thank you

**Pull request closed without merging**